core/src/main/java/org/pentaho/di/connections/vfs/VFSConnectionProvider.java

```diff
@@ -57,4 +57,36 @@ default FileObject getDirectFile( T connectionDetails, String path ) throws Kett
     return KettleVFS.getFileObject( pvfsUrl, new Variables(), getOpts( connectionDetails ) );
   }
 
+  /**
+   * Tests if a given VFS connection is valid, optionally, with certain testing options.
+   *
+   * @param connectionDetails The VFS connection
+   * @param connectionTestOptions The testing options, or {@code null}.
+   * The default implementation exists, call other versions of method.
+   * @return {@code true} if the provided rootPath is valid; {@code false} otherwise.
+   */
+  default boolean test( T connectionDetails, VFSConnectionTestOptions connectionTestOptions ) throws KettleException {
```

Comment:
Could you please use the `@NonNull` and `@Nullable` annotations as suggested in https://github.com/pentaho/pentaho-kettle/pull/9305#discussion_r1595496535?